docs/references/api_endpoints.txt

```diff
@@ -0,0 +1,3229 @@
+        - ``AlreadyAttachedError``: Raised if running on a machine which is already attached to a Pro subscription.
+        - ``AutoAttachDisabledError``: Raised if ``disable_auto_attach: true`` in ``uaclient.conf``.
+        - ``ConnectivityError``: Raised if it is not possible to connect to the contracts service.
+        - ``ContractAPIError``: Raised if there is an unexpected error in the contracts service interaction.
+        - ``EntitlementsNotEnabledError``: Raised if the Client fails to enable any of the entitlements (whether present in any of the lists or listed in the contract).
+        - ``LockHeldError``: Raised if another Client process is holding the lock on the machine.
+        - ``NonAutoAttachImageError``: Raised if the cloud where the system is running does not support auto-attach.
+
+   .. tab-item:: CLI interaction
+      :sync: CLI
+
+      - Calling from the CLI:
+
+        .. code-block:: bash
+
+           pro api u.pro.attach.auto.full_auto_attach.v1 --data {"enable": ["esm-infra", "esm-apps"]}
+
+        
+
+      - Expected attributes in JSON structure:
+
+        .. code-block:: js
+
+           {}
+
+
+
+
+u.pro.attach.auto.should_auto_attach.v1
+=======================================
+
+This endpoint checks if a given system should run auto-attach on boot.
+
+- Introduced in Ubuntu Pro Client Version: ``27.11~``
+
+Arguments:
+
+- This endpoint takes no arguments.
+
+.. tab-set::
+
+   .. tab-item:: Python API interaction
+      :sync: python
+
+      - Calling from Python code:
+
+        .. code-block:: python
+
+           from uaclient.api.u.pro.attach.auto.should_auto_attach.v1 import should_auto_attach
+
+           result = should_auto_attach()
+
+      - Expected return object:
+
+        - ``uaclient.api.u.pro.attach.auto.should_auto_attach.v1.ShouldAutoAttachResult``
+
+          .. list-table::
+             :header-rows: 1
+
+             * - Field Name
+               - Type
+               - Description
+             * - ``should_auto_attach``
+               - ``bool``
+               - True if the system should run auto-attach on boot
+
+      - Raised exceptions:
+
+        - ``UbuntuProError``: ``UbuntuProError`` is the base class of all exceptions raised by Ubuntu Pro Client and it is best practice to handle this error on any API call. (Note: if any API call raises an exception that does not inherit from ``UbuntuProError``, please report a bug).
+
+   .. tab-item:: CLI interaction
+      :sync: CLI
+
+      - Calling from the CLI:
+
+        .. code-block:: bash
+
+           pro api u.pro.attach.auto.should_auto_attach.v1
+
+        
+
+      - Expected attributes in JSON structure:
+
+        .. code-block:: js
+
+           {
+               "should_auto_attach": false
+           }
+
+
+
+
+u.pro.attach.magic.initiate.v1
+==============================
+
+This endpoint initiates the Magic Attach flow, retrieving the User Code to
```

Comment:
I'm not sure how I feel about us referring to the "Magic Attach" process since this is an internal shorthand and not documented anywhere. I know it's not part of this PR necessarily, but it's reminded me we should address this and change to a name that reflects what the process is/is doing.

Reply:
This line didn't change in this PR so I'm not inclined to change it right now. Also, for someone working with the API, "Magic Attach" is the official name (we just don't call it that at the user-facing CLI level).